cfme/ansible/credentials.py

```diff
@@ -0,0 +1,340 @@
+# -*- coding: utf-8 -*-
+"""Page model for Automation/Anisble/Credentials"""
+from navmazing import NavigateToAttribute, NavigateToSibling
+from widgetastic.exceptions import NoSuchElementException
+from widgetastic.utils import ParametrizedLocator
+from widgetastic.widget import ConditionalSwitchableView, ParametrizedView, Text, TextInput, View
+from widgetastic_manageiq import SummaryTable, Table
+from widgetastic_patternfly import BootstrapSelect, Button, Dropdown, FlashMessages, Input
+
+from cfme.base import Server
+from cfme.base.login import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigator, navigate_to, CFMENavigateStep
+from utils.wait import wait_for
+
+
+class CredentialsBaseView(BaseLoggedInPage):
+    flash = FlashMessages('.//div[starts-with(@class, "flash_text_div") or @id="flash_text_div"]')
+    title = Text(locator=".//div[@id='main-content']//h1")
+
+    @property
+    def in_ansible_credentials(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ["Automation", "Ansible", "Credentials"]
+        )
+
+
+class CredentialsListView(CredentialsBaseView):
+    configuration = Dropdown("Configuration")
+    credentials = Table(".//div[@id='list_grid']/table")
+
+    @property
+    def is_displayed(self):
+        return self.in_ansible_credentials and self.title.text == "Credentials"
+
+
+class CredentialDetailsView(CredentialsBaseView):
+    configuration = Dropdown("Configuration")
+    download = Button(title="Download summary in PDF format")
+    properties = SummaryTable("Properties")
+    relationships = SummaryTable("Relationships")
+    credential_options = SummaryTable("Credential Options")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_ansible_credentials and
+            self.title.text == "{} (Summary)".format(self.context["object"].name)
+        )
+
+
+class CredentialFormView(CredentialsBaseView):
+    name = Input(name="name")
+    credential_form = ConditionalSwitchableView(reference="credential_type")
+
+    @credential_form.register("<Choose>", default=True)
+    class CredentialFormDefaultView(View):
+        pass
+
+    @credential_form.register("Machine")
+    class CredentialFormMachineView(View):
+        username = Input(locator='.//input[@title="Username for this credential"]')
+        password = Input(locator='.//input[@title="Password for this credential"][2]')
+        private_key = TextInput(
+            locator='.//textarea[@title="RSA or DSA private key to be used instead of password"][2]'
+        )
+        private_key_phrase = Input(
+            locator='.//input[@title="Passphrase to unlock SSH private key if encrypted"][2]')
+        privilage_escalation = BootstrapSelect("{{name}}")
```

Comment:
yeah, it really has such name. I suspect this has to be resolved to appropriate name. Don't you want to submit a ticket ?